lib/classes/Swift/Transport/StreamBuffer.php

```diff
@@ -87,7 +87,11 @@ public function setParam($param, $value)
 
     public function startTLS()
     {
-        return stream_socket_enable_crypto($this->stream, true, STREAM_CRYPTO_METHOD_TLS_CLIENT);
+        // STREAM_CRYPTO_METHOD_TLS_CLIENT only allow tls1.0 connections (some php versions) 
+        // To support modern tls we allow explicit tls1.0, tls1.1, tls1.2
+        // Ssl3 and older are not allowed because they are vulnerable
+        // @TODO make tls arguments configurable
+        return stream_socket_enable_crypto($this->stream, true, STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT);
```

Comment:
The second argument of `stream_socket_enable_crypto` is now optional. In that case, it takes whatever was configured on the stream context. So, wouldn't it be "better" to remove the second argument here, set a default on the stream when it's created and let the user configure it via `setParam()` like other stream context options?

Reply:
I think `stream_socket_enable_crypto` overwrites the settings of steam context, right?
So `stream_socket_enable_crypto($this->stream, true);` would be the correct code if we would use `setParms`. But then a documentation about enabeling tls1.1/1.2 for php7.0 and 7.1 should be added and thats a bigger patch and not this fast fix..

BTW with php7.2 they reverted STREAM_CRYPTO_METHOD_TLS_CLIENT to all versions of tls1.X, so the problem is only in php7.0 and 7.1